com.unity.visualeffectgraph/Editor/Core/VFXLibrary.cs

```diff
@@ -489,21 +482,66 @@ private static void LoadSRPBindersIfNeeded()
             }
         }
 
+        private static bool unsupportedSRPWarningIssued = false;
+
+        private static void LogUnsupportedSRP(VFXSRPBinder binder, bool forceLog)
+        {
+            if (binder == null && (forceLog || !unsupportedSRPWarningIssued))
+            {
+                Debug.LogWarning("The Visual Effect Graph is supported in the High Definition Render Pipeline (HDRP) and the Universal Render Pipeline (URP). Please assign your chosen Render Pipeline Asset in the Graphics Settings to use it.");
+                unsupportedSRPWarningIssued = true;
+            }
+        }
+
+        public static void LogUnsupportedSRP(bool forceLog = true)
+        {
+            bool logIssued = unsupportedSRPWarningIssued;
+            var binder = currentSRPBinder;
+
+            if (logIssued || !unsupportedSRPWarningIssued) // Don't reissue warning if inner currentSRPBinder call has already logged it
+                LogUnsupportedSRP(binder, forceLog);
+        }
+
         public static VFXSRPBinder currentSRPBinder
         {
             get
             {
                 LoadSRPBindersIfNeeded();
+
                 VFXSRPBinder binder = null;
-                srpBinders.TryGetValue(GraphicsSettings.currentRenderPipeline == null ? "None" : GraphicsSettings.currentRenderPipeline.GetType().Name, out binder);
+                if (GraphicsSettings.currentRenderPipeline != null)
```

Comment:
you should also check against QualitySettings.currentPipelineAsset

if you only access this value at runtime, there are simpler ways to check this with RenderPipelineManager.currentPipeline

Reply:
Actually, this code is purely editor, so, we should use this code :
```
var srpAsset = QualitySettings.renderPipeline ?? GraphicsSettings.renderPipelineAsset;
```
Is it correct @jenniferd-unity ?

Reply:
I'm not sure I understand. How would we test the currentPipeline ? GraphicsSettings or QualitySettings ?

Reply:
what Paul suggested is the correct version. basically if we have an SRP in the QualitySettings it takes precedence over the one assigned in GraphicsSettings.

Reply:
Correct me if I'm wrong :
QualitySettings.renderPipeline returns the current active renderpipeline but it can be null.
If null, GraphicsSettings.renderPipelineAsset (default fallback) will be the active renderpipeline.

The problem of accessing `RenderPipelineManager.currentPipeline` during loading/importing/domain reload is it can returns null if it's done too early.

Reply:
actually no :) almost that

QualitySettings.renderPipelineAsset returns the serialized reference to an SRP Asset of the ACTIVE Quality Level
GraphicsSettings.currentRenderPipeline returns the serialized reference to an SRP Asset of the Graphics Settings
RenderPipelineManager.currentPipeline returns the instance of a Render Pipeline used by the Render Loop (this is the result of the combination of Quality + Graphics). this one is only initialized during the first rendered frame

Reply:
active == in use by the renderer

Reply:
Ok I will check like Paul suggested then

Reply:
you also have `HDRenderPipeline.currentAsset` i forgot about it